includes/class-bsf-custom-fonts-admin.php

```diff
@@ -160,37 +203,175 @@ public function add_new_taxonomy_data() {
+			<?php	} ?>
+			<input type="hidden" name="repeater-field-count" value="<?php echo esc_attr( self::$edit_repeater_field_count ); ?>">
+			<?php wp_nonce_field( basename( __FILE__ ), 'bsf_custom_font_nonce' ); ?>
+			</div>
+			<?php
+		}
+
 		/**
 		 * Edit Taxonomy data
 		 *
 		 * @since 1.0.0
 		 * @param object $term taxonomy terms.
 		 */
-		public function edit_taxonomy_data( $term ) {
-
+		public function edit_new_taxonomy_default_data( $term ) {
 			$data = Bsf_Custom_Fonts_Taxonomy::get_font_links( $term->term_id );
-			$this->font_file_edit_field( 'font_woff_2', __( 'Font .woff2', 'custom-fonts' ), $data['font_woff_2'], __( 'Upload the font\'s woff2 file or enter the URL.', 'custom-fonts' ) );
-			$this->font_file_edit_field( 'font_woff', __( 'Font .woff', 'custom-fonts' ), $data['font_woff'], __( 'Upload the font\'s woff file or enter the URL.', 'custom-fonts' ) );
-			$this->font_file_edit_field( 'font_ttf', __( 'Font .ttf', 'custom-fonts' ), $data['font_ttf'], __( 'Upload the font\'s ttf file or enter the URL.', 'custom-fonts' ) );
-			$this->font_file_edit_field( 'font_eot', __( 'Font .eot', 'custom-fonts' ), $data['font_eot'], __( 'Upload the font\'s eot file or enter the URL.', 'custom-fonts' ) );
-			$this->font_file_edit_field( 'font_svg', __( 'Font .svg', 'custom-fonts' ), $data['font_svg'], __( 'Upload the font\'s svg file or enter the URL.', 'custom-fonts' ) );
-			$this->font_file_edit_field( 'font_otf', __( 'Font .otf', 'custom-fonts' ), $data['font_otf'], __( 'Upload the font\'s otf file or enter the URL.', 'custom-fonts' ) );
-
-			$this->select_edit_field(
+			$this->font_edit_field( 'font_fallback', __( 'Font Fallback', 'custom-fonts' ), __( 'Add the font\'s fallback names with comma(,) separator.  eg. Arial, Serif', 'custom-fonts' ), $data['font_fallback'] );
+
+			$this->select_default_edit_field(
 				'font-display',
 				__( 'Font Display', 'custom-fonts' ),
-				$data['font-display'],
 				__( 'Select font-display property for this font', 'custom-fonts' ),
 				array(
 					'auto'     => 'Auto',
 					'block'    => 'Block',
 					'swap'     => 'Swap',
 					'fallback' => 'Fallback',
 					'optional' => 'Optional',
-				)
+				),
+				$data['font-display']
 			);
 		}
 
+		/**
+		 * Edit Taxonomy data
+		 *
+		 * @since x.x.x
+		 * @param string $key Font array repeater fields key.
+		 * @param string $value Font array repeater fields value.
+		 */
+		public function edit_taxonomy_repeater_data( $key, $value ) {
+			if ( strpos( $key, 'font-weight' ) !== false ) {
+				?>
+				<div id="<?php echo esc_attr( 'item-' . self::$edit_repeater_field_count ); ?>" class="cf-bsf-items" data-group="font-weight-type">
+						<!-- Repeater Content -->
+						<div class="item-content">
+							<div class="form-group">
+							<?php
+							$this->select_new_field(
+								$key,
+								__( 'Font weight', 'custom-fonts' ),
+								__( 'Select font-weight property for this font', 'custom-fonts' ),
+								array(
```

Comment:
I see we have defined this Font Weight array at a few places, so shall we keep them in some static variable to avoid changes in multiple places?

Reply:
@vrundakansara This is only used twice so I think we can keep it as it is. Please let me know your thoughts.

Reply:
Also, the values are different for both the arrays - 'font-weight-0' and `$key`